internal/repository/repository.go

```diff
@@ -252,12 +322,51 @@ func (r *Repository) LookupBlobSize(id restic.ID, tpe restic.BlobType) (uint, bo
 	return r.idx.LookupSize(restic.BlobHandle{ID: id, Type: tpe})
 }
 
+func (r *Repository) getZstdEncoder() *zstd.Encoder {
+	r.allocEnc.Do(func() {
+		level := zstd.SpeedDefault
+		if r.opts.Compression == CompressionMax {
+			level = zstd.SpeedBestCompression
+		}
+
+		enc, err := zstd.NewWriter(nil, zstd.WithEncoderLevel(level))
```

Comment:
Consider if you want `WithEncoderCRC(false)`, if you have checksums elsewhere. It adds a 4 byte xxhash crc.

It is nice to detect corruption, but I am not sure if you actually need it.